pkg/kapp/resources/mod_field_remove.go

```diff
@@ -92,6 +92,19 @@ func (t FieldRemoveMod) apply(obj interface{}, path Path) error {
 			default:
 				panic(fmt.Sprintf("Unknown array index: %#v", part.ArrayIndex))
 			}
+		case part.Regex != nil && part.Regex.Regex != nil:
+			matchedKeys, err := t.obtainMatchingRegexKeys(obj, part)
+			if err != nil {
+				return err
+			}
+			for _, key := range matchedKeys {
+				newPath := append(Path{&PathPart{MapKey: &key}}, path[i+1:]...)
+				err := t.apply(obj, newPath)
+				if err != nil {
+					return err
+				}
+			}
+			return nil
```

Comment:
We can probably get rid of the `obtainMatchingRegexKeys` by combining some stuff (as it just seems to be moving the complexity from one place to another):
```
			regex, err := regexp.Compile(*part.Regex.Regex)
			if err != nil {
				return err
			}

			typedObj, ok := obj.(map[string]interface{})
			if !ok && typedObj != nil {
				return fmt.Errorf("Unexpected non-map found: %T", obj)
			}

			for key := range typedObj {
				if regex.MatchString(key) {
					newPath := append(Path{&PathPart{MapKey: &key}}, path[i+1:]...)
					err := t.apply(obj, newPath)
					if err != nil {
						return err
					}
				}
			}
			return nil
```

Reply:
Yesh I can that as well just abstracted the logic in different function to convey what exactly we are trying to do and if we want to rid of this function from here should we do the same in `mod_field_copy.go` as well?

Reply:
> If we want to rid of this function from here then we should have to do for mod_field_copy.go as well.

We probably can.